src/Surface.zig

```diff
@@ -3582,18 +3580,49 @@ pub fn mouseButtonCallback(
             break :pin pin;
         };
 
-        // If we already have a selection and the selection contains
-        // where we clicked then we don't want to modify the selection.
-        if (self.io.terminal.screen.selection) |prev_sel| {
-            if (prev_sel.contains(screen, pin)) break :sel;
+        switch (self.config.right_click_action) {
+            .ignore => {
+                // Return early to skip clearing the selection.
+                try self.queueRender();
+                return true;
+            },
+            .copy => {
+                if (self.io.terminal.screen.selection) |sel| {
+                    copySelectionToClipboards(self, sel, &.{.standard});
+                }
+            },
+            .@"copy-or-paste" => {
+                if (self.io.terminal.screen.selection) |sel| {
+                    copySelectionToClipboards(self, sel, &.{.standard});
+                } else {
+                    try self.startClipboardRequest(.standard, .{ .paste = {} });
```

Comment:
A tagged union variant with type `void` can be more simply written as an enum literal
```suggestion
                    try self.startClipboardRequest(.standard, .paste);
```